src/synapse/azext_synapse/custom.py

```diff
@@ -153,27 +179,118 @@ def create_spark_pool(cmd, client, resource_group_name, workspace_name, spark_po
         big_data_pool_info.auto_pause = AutoPauseProperties(enabled=auto_pause_enabled,
                                                             delay_in_minutes=delay_in_minutes)
 
-    if library_requirements_filename or library_requirements_content:
-        big_data_pool_info.library_requirements = LibraryRequirements(filename=library_requirements_filename,
+    if library_requirements_file:
+        library_requirements_content = read_file_content(library_requirements_file)
+        big_data_pool_info.library_requirements = LibraryRequirements(filename=library_requirements_file,
                                                                       content=library_requirements_content)
     return sdk_no_wait(no_wait, client.create_or_update, resource_group_name, workspace_name, spark_pool_name,
                        big_data_pool_info, force)
 
 
-def update_spark_pool(cmd, client, resource_group_name, workspace_name, spark_pool_name, tags=None):
+def update_spark_pool(cmd, client, workspace, spark_pool_name, tags=None):
+    parsed_workspace = extract_workspace_and_resource_group(workspace)
+    workspace_name = parsed_workspace['name']
+    resource_group_name = parsed_workspace['resource_group']
     return client.update(resource_group_name, workspace_name, spark_pool_name, tags)
 
 
-def delete_spark_pool(cmd, client, resource_group_name, workspace_name, spark_pool_name, no_wait=False):
+def delete_spark_pool(cmd, client, workspace, spark_pool_name, no_wait=False):
+    parsed_workspace = extract_workspace_and_resource_group(workspace)
+    workspace_name = parsed_workspace['name']
+    resource_group_name = parsed_workspace['resource_group']
     return sdk_no_wait(no_wait, client.delete, resource_group_name, workspace_name, spark_pool_name)
 
 
-def create_sql_pool(cmd, client, resource_group_name, workspace_name, sql_pool_name, location, sku_name,
-                    sku_tier=None, max_size_bytes=None, source_database_id=None, recoverable_database_id=None,
+def get_sql_pool(cmd, client, workspace, sql_pool_name):
+    parsed_workspace = extract_workspace_and_resource_group(workspace)
+    workspace_name = parsed_workspace['name']
+    resource_group_name = parsed_workspace['resource_group']
+    return client.get(resource_group_name, workspace_name, sql_pool_name)
+
+
+def list_sql_pool_by_workspace(cmd, client, workspace):
+    parsed_workspace = extract_workspace_and_resource_group(workspace)
+    workspace_name = parsed_workspace['name']
+    resource_group_name = parsed_workspace['resource_group']
+    return client.list_by_workspace(resource_group_name, workspace_name)
+
+
+def create_sql_pool(cmd, client, workspace, sql_pool_name, sku_name, source_database_id=None,

```

Comment:
We might need to determine create_mode internally. Could you please reference some exisiting SQL cmdlets? For example: 
1. `az sql db create`
2. `az sql db restore`

Reply:
> We might need to determine create_mode internally. Could you please reference some exisiting SQL cmdlets? For example:
> 
> 1. `az sql db create`
> 2. `az sql db restore`

`az sql db create` doesn't expose create_mode. So removed it